vowpalwabbit/plt.cc

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#include <cfloat>
+#include <cmath>
+#include <cstdio>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <algorithm>
+#include <queue>
+#include "reductions.h"
+#include "vw.h"
+
+using namespace VW::LEARNER;
+using namespace VW::config;
+
+namespace plt_ns
+{
+struct node
+{
+  uint32_t n;  // node number
+  float p;     // node probability
+
+  bool operator<(const node &r) const { return p < r.p; }
+};
+
+struct plt
+{
+  vw *all;
+
+  uint32_t k;     // number of labels
+  uint32_t t;     // number of tree nodes
+  uint32_t ti;    // number of internal nodes
+  uint32_t kary;  // kary tree
+
+  v_array<float> nodes_t;  // in case of sgd, this keeps stores individual t for each node
```

Comment:
Nit: `_t` implies this is a type and not a member variable.

Reply:
I changed the name to omit the confusion.